app/controllers/post_controller.rb

```diff
@@ -1,59 +1,52 @@
 class PostController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_post, only: [:show, :show_draft, :update, :destroy]
+  expose :post, scope: ->{ current_user.posts }
+  expose :comment, fetch: ->{ Comment.new }
+  expose :comments, fetch: ->{ Comment.where(post_id: post.id) }
 
   def new
-    @post = current_user.posts.new
   end
 
   def create
-    @post = Post.new(post_params)
-    @post.user_id = current_user.id
-
     if params["commit"] == "Create Post"
-      @post.done = true
+      post.done = true
     else
-      @post.done = false
+      post.done = false
     end
 
-    @post.save
+    post.save
     redirect_to root_path
   end
 
   def show
-    @comment = Comment.new
-    @comments = Comment.where(post_id: @post.id)
   end
 
   def show_draft
```

Comment:
in controller must not be any other function except default CRUD: new create edit update destroy

If you want to show "draft" it means you have other kind of object and you need controller for it, for example DraftPostsController#show